build.zig

```diff
@@ -1,18 +1,26 @@
 const std = @import("std");
 
 pub fn build(b: *std.Build) void {
-    // TODO Proper library build (how to retain the ability to make an executable for development?)
-    // Though honestly I think the easiest way to use this library is to just copy it into your project.
-
     const target = b.standardTargetOptions(.{});
     const optimize = b.standardOptimizeOption(.{});
 
-    const exe = b.addExecutable(.{
-        .name = "FastNoiseLite",
-        .root_source_file = .{ .path = "test.zig" },
+    // Exposing module to the build system so the dependees can use it.
+    _ = b.addModule("fast-noise-lite", .{ .root_source_file = b.path("src/fast_noise_lite.zig") });
+
+    const lib = b.addStaticLibrary(.{
+        .name = "fast-noise-lite-zig",
+        .root_source_file = b.path("src/fast_noise_lite.zig"),
         .optimize = optimize,
         .target = target,
     });
+    b.installArtifact(lib);
```

Comment:
I'm wondering, what is the point of a static library (and specifying again the root source file and an extra name) when there is already a module? What will actually be used by Zig projects? Is this for different ways of depending on it? The code is almost entirely generics so I'm not sure what a static lib would provide.

Reply:
The static lib is necessary for linkage and necessary for niche edge cases where people might link it to c/c++ projects or anything else supporting C ABI interop.